### PR TITLE
addressing java migration technical debt

### PR DESCRIPTION
**Pull request by foman-ch**
JIRA: https://companieshouse.atlassian.net/browse/JU-529

upgrading Spring Boot to 3.3.2
upgrading dependency versions
implementation of new suppressions method

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Review by aduffy-ch — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -74,28 +74,31 @@
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-actuator</artifactId>
-            <version>3.2.4</version>
         </dependency>
         <dependency>
-            <!--Overriding spring-web version to remove vulnerability-->
```
Comment (vshastri-ch) [inline pom.xml:80]: Could remove this?

**Review by vshastri-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request merged**